src/web/rustdoc.rs

```diff
@@ -795,6 +798,29 @@ impl Handler for SharedResourceHandler {
     }
 }
 
+/// Serves shared web resources used by rustdoc-generated documentation.
+///
+/// Rustdoc has certain JS, CSS, font and image files that are required for all
+/// documentation it generates, and these don't change often. We make one copy
+/// of these per rustdoc release and serve them from a common location.
+///
+/// This handler considers the whole path, and looks for a file at that path in
+/// the Storage.
+pub struct SharedResourceHandler;
+
+impl Handler for SharedResourceHandler {
+    fn handle(&self, req: &mut Request) -> IronResult<Response> {
+        let storage = extension!(req, Storage);
+        let config = extension!(req, Config);
+
+        let storage_path = format!("/{}", req.url.path().join("/"));
+        match File::from_path(storage, &storage_path, config) {
+            Ok(file) => Ok(file.serve()),
+            Err(_) => Err(Nope::ResourceNotFound.into()),
```

Comment:
nit: we shouldn't ignore server errors here, but report them. 

Something like:
```rust 
    match File::from_path(storage, filename, config) {
        Ok(file) => Ok(file.serve()),
        Err(err) if err.downcast_ref::<PathNotFoundError>().is_some() => {
            Err(Nope::ResourceNotFound.into())
        }
        Err(err) => {
            utils::report_error(&err);
            Err(Nope::InternalServerError.into())
        }
    }
```